spec/features/admin/basic_project_flow_spec.rb

```diff
@@ -14,4 +14,12 @@
     subject { basic_project }
     let(:edit_button_name) { 'Edit Project' }
   end
+
+  scenario 'basic project can not be created with the same person as pry and sec agent' do
+    visit new_admin_basic_project_path
+    select user.name, from: 'basic_project_primary_agent_id'
+    select user.name, from: 'basic_project_secondary_agent_id'
+    click_on 'Create Basic project'
+    expect(page).to have_content('The primary agent for this project cannot be the same as the secondary agent')
+  end
```

Comment:
@arisah-ijeoma Good job adding a spec, but typically you don't put specs that test such specific bit model functionality in a feature spec. That's what model specs are for. See the testing pyramid:

https://martinfowler.com/bliki/TestPyramid.html

Feature specs should mostly focus on the happy path, while model specs should cover all the nitty gritty details.

Luckily we have model specs for these validations already.

So normally I'd say to remove these feature specs but on the other hand we don't have _any_ coverage of the validation error flow. How about you just rename the scenarios to "validation", and we can use the agent stuff as a test case for whether general validation works right. Calling the scenario "validation" will be a signifier that this is intended as a generic validation flow spec instead of a specific test of this specific validation.

In future though, no need to add more feature specs if, say, additional validation rules are added. Just model specs are sufficient.

If any of this is confusing, let me know.

Thanks!

Reply:
thank you for this insightful detail @hooverlunch